internals/cli/cmd_exec.go

```diff
@@ -223,6 +223,22 @@ func (cmd *cmdExec) Execute(args []string) error {
 	}
 }
 
+// willMakeTerminalRaw returns true if the exec command will make the
+// terminal raw during execution.
+func (cmd *cmdExec) willMakeTerminalRaw() bool {
```

Comment:
Most of the lines of this function are repeated in `cmdExec.Execute`. Could we create a function `useTerminal` that does most of this, and is used in both places?

```go
func (cmd *cmdExec) useTerminal(stdoutIsTerminal bool) bool {
	if cmd.Terminal {
		return true
	} else if cmd.NoTerminal {
		return false
	} else {
		return stdoutIsTerminal
	}
}
```

And then we could get rid of `willMakeTerminalRaw`, or it'd be a simple 3-liner.

Alternatively we could just remove this from `cmd_exec.go` and inline it in `cmd_enter.go` -- it seems weird to me that it's defined here but not used here.

Reply:
Thanks! I added the `useTerminal` function and removed `willMakeTerminalRaw`. Changes are in https://github.com/canonical/pebble/pull/347/commits/5f1924635b68917bf1d0d479f1824c0be631d064.